client/components/site-preview-link/use-site-preview-links.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import type { SiteId } from 'calypso/types';
+
+export interface PreviewLink {
+	code: string;
+	created_at: string;
+	isCreating?: boolean;
+	isRemoving?: boolean;
+}
+
+export type PreviewLinksResponse = PreviewLink[];
+
+interface UseSitePreviewLinksOptions {
+	siteId: SiteId;
+	onSuccess?: ( previewLinks: PreviewLink[] | undefined ) => void;
+}
+
+export const SITE_PREVIEW_LINKS_QUERY_KEY = 'site-preview-links';
+
+export function useSitePreviewLinks( options: UseSitePreviewLinksOptions ) {
+	const { siteId, onSuccess } = options;
+	return useQuery< PreviewLinksResponse, unknown, PreviewLink[] >(
+		[ SITE_PREVIEW_LINKS_QUERY_KEY, siteId ],
+		() =>
+			wpcom.req.get( {
+				path: `/sites/${ siteId }/preview-links`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		{
+			enabled: !! siteId,
+			meta: { persist: false },
+			onSuccess,
```

Comment:
👋🏻 @sejas I'm reaching out because this `onSuccess` handler is a blocking part of the [RQv5 update](https://github.com/Automattic/wp-calypso/pull/84338#issuecomment-1824525840) – `on*` handlers are removed from the `useQuery` API ([see this note](https://tanstack.com/query/latest/docs/react/guides/migrating-to-v5#callbacks-on-usequery-and-queryobserver-have-been-removed)). Would you be able to help or know someone who is familiar with this code & functionality and would be able help with the migration to v5?

Reply:
Sure!, I can take a look and create a PR.

Reply:
Thank you!